jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -422,24 +506,38 @@ object XGBoost extends Serializable {
       case "_reg_" =>
         val xgBoostModel = new XGBoostRegressionModel(SXGBoost.loadModel(dataInStream))
         setGeneralModelParams(featureCol, labelCol, predictionCol, xgBoostModel)
+      case other =>
+        throw new XGBoostError(s"Unknown model type $other. Supported types " +
+          s"are: ['_reg_', '_cls_'].")
     }
   }
 }
 
+/** Class to hold the training and test matrices in a given partition. */
 private class Watches private(val train: DMatrix, val test: DMatrix) {
-  def toMap: Map[String, DMatrix] = Map("train" -> train, "test" -> test)
-      .filter { case (_, matrix) => matrix.rowNum > 0 }
+
+  def toMap: Map[String, DMatrix] = {
+    Map("train" -> train, "test" -> test).filter { case (_, matrix) => matrix.rowNum > 0 }
+  }
 
   def size: Int = toMap.size
 
-  def delete(): Unit = {
-    toMap.values.foreach(_.delete())
-  }
+  def delete(): Unit = toMap.values.foreach(_.delete())
 
   override def toString: String = toMap.toString
 }
 
 private object Watches {
+
+  /**
+   * Build train and test `DMatrix` from the Spark partition iterator.
+   *
+   * @param params Parameter map.
+   * @param labeledPoints Iterator of [[XGBLabeledPoint]].
+   * @param baseMarginsOpt Option containing base margins for each labeled point.
+   * @param cacheFileName Prefix for data cache files.
+   * @return Watches instance, holding train and test matrices.
+   */
```

Comment:
is it so hard to recognize this method is `Build train and test `DMatrix` from the Spark partition iterator.`?

Reply:
I'm not sure what you mean. Just for reference, I added these docs based off the perspective of someone who is new to the codebase (which I was), so I do think they are helpful. Here in particular I don't understand why the class is called `Watches` and I was confused when I initially encountered it. 

In general, of course you can understand what the code does by reading it. It's a matter of time investment. I added these to save others time in the future. I spent a reasonably large chunk of time learning this code because I was trying to debug errors I got while running an example, and saving 60 seconds here and there would have been very useful, yes. 

Honestly, I don't understand the resistance to adding short docs here and there that can help future developers. I'm sure it is obvious to those who wrote the code, but it was not obvious to me.